R/filter2.R

```diff
@@ -45,10 +45,21 @@ filter2 = function(x, filter, boundary = c("circular", "replicate")) {
          },
          ## pad with a given value
          linear = {
-           dx[1:2] = dx[1:2] + cf[1:2]
-           xpad = array(val, dx)
-           ## is there a better way of doing this?
-           x = do.call("[<-", c(quote(xpad), lapply(d, function(x) enquote(1:x)), quote(x)) )
+           x = imageData(x)
+           
+           dx[1:2] = dx[1:2] + df[1:2] - 1L
+           
+           if ( length(dx)>2 & length(val)==prod(dx[-(1:2)]) ) {
+             # Higher dim array with matching linear boundry values
+             xpad = array(rep(val, each=prod(dx[1:2])), dim = dx)
+           } else {
+             if ( length(val)>1 ) {
+               warning('The boundary value length does not match the number of frames, only the first element of boundary will be used')
+             }
+             xpad = array(val[1], dx)
+           }
+           # The do.call and arguments as a list of dimension is to take are of arrays of unknown dimension
+           x = do.call("[<-", c(list(xpad),lapply(d,FUN = function(x) 1:x),list(x)) )
```

Comment:
I took your comment to hart: ## is there a better way of doing this?  I didn't think you needed the quotes and were just costing overhead. 

I have looked at http://adv-r.had.co.nz/Computing-on-the-language.html for some inspiration, but I cannot think of a use case where quotes or enquote would be needed. It also debugs just fine.

All but the last argument cannot be formulas or expressions (they are always arrays/matrices or vectors). We could perhaps quote the last argument `x`. But not sure how you would use that.